File: 242-valid-anagram/valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()){
            return false;
            }
        int len = s.length();
        int[] myArr = new int[26];
        for (int i = 0 ; i < len ; i++){
            myArr[s.charAt(i) - 'a' ]++;
            myArr[t.charAt(i) - 'a' ]--;
        }
        for(int j = 0 ; j < 26 ; j++){
            if(myArr[j] != 0){
                return false;
                }
        }
        return true;
    }
}

File: 242-valid-anagram/valid-anagram.c
bool isAnagram(char* s, char* t) {
    //counting array solution, one increment and other decrement , when all values are 0 its anagram
    int myArr[26] = {0};
    if (strlen(t) != strlen(s)){
        return false;
    }
    for (int i = 0 ; i < strlen(s) ; i++){
        myArr[s[i] - 'a']++;
        myArr[t[i] - 'a']--;
    }
    for(int j = 0 ; j <26 ; j++){
        if( myArr[j] != 0 ){
            return false;
        }

    }
    return true;
}